src/net.cpp

```diff
@@ -2642,16 +2651,69 @@ void CConnman::RecordBytesRecv(uint64_t bytes)
     nTotalBytesRecv += bytes;
 }
 
-void CConnman::RecordBytesRecvByMsgType(mapMsgTypeSize msgtype_bytes)
+void CConnman::RecordMsgStatsRecv(std::map<std::string, std::tuple<int, uint64_t>> msgtype_countbytes,
+    ConnectionType conn_type, Network net_type)
 {
-    for (auto const& msg_type_bytes : msgtype_bytes) {
-        auto i = m_msgtype_bytes_recv.find(msg_type_bytes.first);
+    for (auto const& msgtype_stats : msgtype_countbytes) {
+        std::string msg_type = msgtype_stats.first;
+        auto i = m_msgtype_bytes_recv.find(msg_type);
         if (i == m_msgtype_bytes_recv.end()) {
             i = m_msgtype_bytes_recv.find(NET_MESSAGE_TYPE_OTHER);
         }
 
         assert(i != m_msgtype_bytes_recv.end());
-        i->second += msg_type_bytes.second;
+        std::tuple<int, uint64_t> count_bytes = msgtype_stats.second;
+
+        int msg_count = std::get<0>(count_bytes);
+        uint64_t num_bytes = std::get<1>(count_bytes);
+
+        if (num_bytes > 0) {
+            i->second += num_bytes;
+
+            NetMsgStatsKey stats_key = {msg_type, conn_type, net_type};
+
+            LogPrint(BCLog::NET, "\nstacie - incremeting received message count by %d, num_bytes by %d for (%s, %s, %s)\n",
+                msg_count, num_bytes, msg_type, ConnectionTypeAsString(conn_type), NetworkAsString(net_type));
+            auto stats_iterator = m_netmsg_stats_recv.find(stats_key);
+
+            // If this stat does not exist, add it (could also initialize all stats to zero?)
+            if (stats_iterator == m_netmsg_stats_recv.end()) {
+                LogPrint(BCLog::NET, "\nstacie - creating a new entry in the map\n");
+                NetMsgStatsValue stats_value = {msg_count, num_bytes};
+                m_netmsg_stats_recv.insert(std::map<NetMsgStatsKey, NetMsgStatsValue>::value_type(stats_key, stats_value));
+            } else {
+                // otherwise, update the message count and number of bytes
+                LogPrint(BCLog::NET, "\nstacie - adding to existing entry instead of making a new one\n");
+                stats_iterator->second.msg_count += msg_count;
+                stats_iterator->second.num_bytes += num_bytes;
+            }
+        }
+    }
+}
+
+void CConnman::PrintNetMsgStats() const {
```

Comment:
Write an aggregator so I don't have to keep spot checking this with what getnettotals returns